jung-visualization/src/main/java/edu/uci/ics/jung/visualization/subLayout/GraphCollapser.java

```diff
@@ -71,76 +73,110 @@ public Network collapse(Network inGraph, Network clusterGraph) {
     return graph;
   }
 
-  public Network expand(Network inGraph, Network clusterGraph) {
-    MutableNetwork graph = graphBuilder.build();
-    Collection clusterNodes = clusterGraph.nodes();
-    logger.fine("cluster to expand is " + clusterNodes);
-
-    // put all clusterGraph vertices and edges into the new Graph
-    for (Object node : clusterNodes) {
-      graph.addNode(node);
-      for (Object edge : clusterGraph.incidentEdges(node)) {
-        Object u = clusterGraph.incidentNodes(edge).nodeU();
-        Object v = clusterGraph.incidentNodes(edge).nodeV();
-        graph.addEdge(u, v, edge);
+  public Network expand(Network originalNetwork, Network inGraph, Network clusterGraphNode) {
+
+    NetworkBuilder networkBuilder = NetworkBuilder.from(originalNetwork);
+    // build a new empty network
+    MutableNetwork newGraph = networkBuilder.build();
+    // add all the nodes from inGraph that are not clusterGraphNode and are not in clusterGraphNode
+    for (Object node : inGraph.nodes()) {
+      if (!node.equals(clusterGraphNode) && !this.contains(clusterGraphNode, node)) {
+        newGraph.addNode(node);
       }
     }
-    // add all the vertices from the current graph except for
-    // the cluster we are expanding
-    for (Object node : inGraph.nodes()) {
-      if (node.equals(clusterGraph) == false) {
-        graph.addNode(node);
+
+    // add all edges that don't have an endpoint that either is clusterGraphNode or is in clusterGraphNode
+    for (Object edge : inGraph.edges()) {
+      EndpointPair endpoints = inGraph.incidentNodes(edge);
+      boolean dontWantThis = false;
```

Comment:
Stylistic suggestion: code is generally clearer if variable names are phrased positively.  In this case I recommend "shouldAdd":

    boolean shouldAdd = true;
    for (Object endpoint : endpoints) {
      shouldAdd &= !(endpoint.equals(clusterGraphNode) || this.contains(clusterGraphNode, endpoint));
    }
    if (shouldAdd) {
      newGraph.addEdge(...);
    }

Reply:
...so a positive LHS paired with a negated RHS is clearer to you?